ui/plugins/eu.esdihumboldt.hale.ui/src/eu/esdihumboldt/hale/ui/io/schema/SourceSchemaActionUIAdvisor.java

```diff
@@ -0,0 +1,72 @@
+
+package eu.esdihumboldt.hale.ui.io.schema;
+
+import org.eclipse.ui.PlatformUI;
+
+import eu.esdihumboldt.hale.common.instance.model.DataSet;
+import eu.esdihumboldt.hale.common.instance.model.Instance;
+import eu.esdihumboldt.hale.common.instance.model.InstanceCollection;
+import eu.esdihumboldt.hale.common.instance.model.ResourceIterator;
+import eu.esdihumboldt.hale.common.schema.SchemaSpaceID;
+import eu.esdihumboldt.hale.common.schema.model.Schema;
+import eu.esdihumboldt.hale.ui.io.action.AbstractActionUIAdvisor;
+import eu.esdihumboldt.hale.ui.service.instance.InstanceService;
+import eu.esdihumboldt.hale.ui.service.schema.SchemaService;
+
+/**
+ * Source Schema action UI advisor.
+ * 
+ * @author Kapil Agnihotri.
+ */
+public class SourceSchemaActionUIAdvisor extends AbstractActionUIAdvisor<Schema> {
```

Comment:
I think there is enough common functionality to introduce an abstract base class with common functionality for the source and target advisors.

Reply:
Introduced an abstract class `AbstractSchemaActionUIAdvisor<T>` as per the suggestion.